scene/animate-images-with-image-overlay/src/main/java/com/esri/samples/animate_images_with_image_overlay/AnimateImagesWithImageOverlayController.java

```diff
@@ -0,0 +1,208 @@
+
+      // create a new elevation source from the Terrain3D REST service and set it as the scene's base surface
+      Surface surface = new Surface();
+      surface.getElevationSources().add(new ArcGISTiledElevationSource("https://elevation3d.arcgis" +
+        ".com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer"));
+      scene.setBaseSurface(surface);
+
+      // create and append an image overlay to the scene view
+      sceneView.getImageOverlays().add(new ImageOverlay());
+
+      // instantiate a new empty list to hold image frames
+      imageFrames = new ArrayList<>();
+      // get the image files from local storage as an unordered list
+      File[] imageFiles = new File(System.getProperty("data.dir"), "./samples-data/PacificSouthWest").listFiles();
+      // sort the list of image files
+      if (imageFiles != null) {
+        Arrays.sort(imageFiles);
+        // create an image with the given path and use it to create an image frame
+        for (File file : imageFiles) {
+          ImageFrame imageFrame = new ImageFrame(file.getAbsolutePath(), imageFrameEnvelope);
+          imageFrames.add(imageFrame);
+        }
+      }
+
+      // populate the frames combo box with values
+      framesComboBox.getItems().addAll("60 frames per second", "30 frames per second", "15 frames per second");
+      // open the sample at 15fps
+      framesComboBox.getSelectionModel().select(2);
+
+      startNewAnimationTimer();
+
+    } catch (Exception e) {
+      // on any exception, print the stack trace
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Create a new image frame from the image at the current index and add it to the image overlay.
+   */
+  private void addNextImageFrameToImageOverlay() {
+
+    // set image frame to image overlay
+    sceneView.getImageOverlays().get(0).setImageFrame(imageFrames.get(imageIndex));
+    // increment the index to keep track of which image to load next
+    imageIndex++;
+    // reset index once all files have been loaded
+    if (imageIndex == imageFrames.size())
+      imageIndex = 0;
+  }
+
+  /**
+   * Controls the opacity of the image overlay using the slider.
+   */
+  @FXML
+  private void changeImageOverlayOpacity() {
+    sceneView.getImageOverlays().get(0).setOpacity((float) opacitySlider.getValue());
+  }
+
+  /**
+   * Set up a timer to display the images at the specified frame rate from the combobox.
+   */
+  private void startNewAnimationTimer() {
+
+    timer = new Timer(true);
+    TimerTask timerTask = new TimerTask() {
+      @Override
+      public void run() {
+        addNextImageFrameToImageOverlay();
+      }
+    };
+    timer.scheduleAtFixedRate(timerTask, 1L, period);
+  }
+
+  /**
+   * Handles the rate at which the image frames are displayed using the combo box.
+   */
+  @FXML
+  private void handleframesComboBoxInteraction() {
+    // set the period for the chosen fps 
+    switch (framesComboBox.getSelectionModel().getSelectedItem()) {
```

Comment:
If you wanted, you could add Key-Value pairs of String-Int to the ComboBox, and use a cell factory to display the key (15/30/60 fps), and then on selection just put in the corresponding value (17/33/67) straight into the `startNewAnimationTimer`. Let me know if you want me to walk you through it.

Reply:
Thanks, I'd considered the use of a cell factory but then decided to go for the switch since it's a fairly basic set up and we can comment to show the conversion "maths". We have other Java samples that use the switch for similar "simplicity" so it seemed fitting. Though we also have cell factory for simple conversions too! Unless you have strong feelings about it we can leave as is.

Reply:
I think the whole conversion stuff distracts from the actual sample's intentions, but if there's other samples that do this too then you can just leave it.